propagation/w3c/RATIONALE.md

```diff
@@ -0,0 +1,62 @@
+# brave-propagation-w3c rationale
+
+## Trace Context Specification
+
+### Why do we write a tracestate entry?
+We write both "traceparent" and a "tracestate" entry for two reasons. The first is due to
+incompatibility between the "traceparent" format and our trace context. This is described in another
+section.
+
+The other reason is durability across hops. When your direct upstream is not the same tracing
+system, its span ID (which they call parentId in section 3.2.2.4) will not be valid: using it will
+break the trace. Instead, we look at our "tracestate" entry, which represents the last known place
+in the same system.
+
+### What is supported by Brave, but not by traceparent format?
+
+#### `SamplingFlags` can precede a trace
+B3 has always had the ability to influence the start of a trace with a sampling decision. For
+example, you can propagate `b3=0` to force the next hop to not trace the request. This is used for
+things like not sampling health checks. Conversely, you can force sampling by propagating `b3=d`
+(the debug flag). `traceparent` requires a trace ID and a span ID, so cannot propagate this.
+
+#### `TraceIdContext` (trace ID only)
+Amazon trace format can propagate a trace ID prior to starting a trace, for correlation purposes.
+`traceparent` requires a trace ID and a span ID, so cannot a trace ID standalone.
+
+#### Not yet sampled/ deferred decision
+B3 and Amazon formats support assigning a span ID prior to a sampling decision. `traceparent` has no
+way to tell the difference between an explicit no and lack of a decision, as it only has one bit
+flag.
+
+#### Debug flag
```

Comment:
TODO: I forgot to add the (actual) parent ID. the traceparent labels span ID parent ID (it is so confusing.. if span was a bad word, just say ID :P). traceparent does not record the parent ID of the context propagated, basically.